android/src/main/java/com/hoxfon/react/RNTwilioVoice/TwilioVoiceModule.java

```diff
@@ -687,6 +745,12 @@ public void accept() {
         if (BuildConfig.DEBUG) {
             Log.d(TAG, "accept()");
         }
+
+        Intent intent = new Intent(getReactApplicationContext(), IncomingCallNotificationService.class);
+        intent.setAction(Constants.ACTION_ANSWER);
+
+        getReactApplicationContext().startService(intent);
```

Comment:
I see what you are doing here. When the JS accept the call you start a service to capture the intent of ending the heads up notification. But is it allowed? I thought about this but I didn't pursue this route as the app is running and it didn't look neat to start a service while the app is running.

Reply:
Yes, and it works for our project)) The idea is when the user accepts or declines a call I just stop foreground service (which is ok because any further actions are made in JS directly and we don't need any pushes)